### DISCUSSION TITLE
Morse Runner on Mac

### DISCUSSION THREAD

**Discussion by scotthibbs** [w7sst/MorseRunner#118] [Ideas]
I've seen a few asking if there will be a Mac version. 

This may be a separate repository or beyond our capability. 

However, I wanted to pass on that this has been working on Macs prior to Catalina here: 

https://www.merzhaus.org/ham/MorseRunner/index.html

**Comment by tekenny** [comment]
I would like to see MR on platforms other than Windows but think it isn’t a priority for now IMHO…

**Reply by w7sst** [reply]
I agree. I added a `someday maybe` label and added it to this issue.

**Comment by w7sst** [comment]
I converted this to an issue #120.

**Comment by w7sst** [comment]
A Morse Runner Mac port is a reality!

Thomas, DJ1TF, has created a fork of the original Morse Runner on a Mac using PortAudio. He recently announced this port on the [MorseRunnerCE user community page](https://groups.io/g/MorseRunnerCE/topic/104131508#87) on groups.io.

I will reach out to Thomas and see if we can start a dialog in this Discussion page about merging his changes into MRCE. His work is based on Lazarus, not Delphi. I am not tied to Delphi and have been considering moving our code base back to Lazarus for some time now.

My time availability will be limited. I'll be available for discussion, code reviews, guidance and code reviews. I would like us to consider this work and see if we can arrive at a strategy to perhaps merge Thomas Mac port into MRCE. We also want to be mindful of Jim's work (K6OK) of an User Interface improvement prototype; I would like to see us move forward on Jim's work as well.

Please join the discussion. Thank you and 73, Mike W7SST

**Comment by w7sst** [comment]
@K6OK 
Jim, see the comments above about a recent fork of the original MR on a Mac. I have reached out to Thomas on groups.io, but have not heard back. The fact that he was able to get the base MR code base running with Lazarus on a Mac is good news. There is another fork from someone who has it running on Linux.

**Reply by K6OK** [reply]
Mike -- I agree, migrating from Delphi to Lazarus is worthy of consideration.  Delphi CE only supports 32- and 64-bit Windows, while Lazarus supports Windows, Apple and Linux.  Also, Delphi's one year licenses add uncertainty.  We don't know for how much longer that will last.
